transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferDestinationActivity.java

```diff
@@ -0,0 +1,109 @@
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.transfer.R;
+import com.hyperwallet.android.ui.transfer.viewmodel.ListTransferDestinationViewModel;
+
+public class ListTransferDestinationActivity extends AppCompatActivity {
+
+    public static final String ARGUMENT_SELECTED_DESTINATION = "argument_selected_destination";
+    private static final String TAG = ListTransferDestinationActivity.class.getSimpleName();
+
+    private ListTransferDestinationViewModel mListTransferDestinationViewModel;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list_transfer_destination);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        toolbar.setTitle(R.string.destination);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+
+        if (savedInstanceState == null) {
+            initFragment(ListTransferDestinationFragment.newInstance(
+                    (HyperwalletTransferMethod) getIntent().getParcelableExtra(ARGUMENT_SELECTED_DESTINATION)));
+        }
+
+        mListTransferDestinationViewModel = ViewModelProviders.of(this).get(
+                ListTransferDestinationViewModel.class);
+
+        registerObservers();
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        onBackPressed();
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        getWindow().setStatusBarColor(ContextCompat.getColor(this, R.color.colorPrimaryDark));
+        getWindow().getDecorView().setSystemUiVisibility(0);
+        super.onBackPressed();
+    }
+
+    private void initFragment(Fragment fragment) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction transaction = fragmentManager.beginTransaction();
+        transaction.add(R.id.frame, fragment);
+        transaction.commit();
+    }
+
+    private void registerObservers() {
+        mListTransferDestinationViewModel.getTransferDestinationSection().observe(this,
```

Comment:
I think the activity needs to also listen for initialization errors and show a dialog to inform the user you have initialization/network error while loading transfer method list

Reply:
Added here https://github.com/hyperwallet/hyperwallet-android-ui-sdk/pull/84